File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.py
'''
아이디어1. 문자열에서 인덱스로 접근 > 인덱스 초과 에러
아이디어2. 스택 이용
'''

def solution(s):

    stack = []

    for i in s:
        if stack:
            if i == stack[-1]:
                stack.pop()
                continue
            else:
                stack.append(i)
        else:
             stack.append(i)   

    return 0 if stack else 1

File: 프로그래머스/2/12973. 짝지어 제거하기/짝지어 제거하기.java
import java.util.*;

class Solution
{
    public int solution(String s)
    {
        Stack<Character> stack = new Stack<>();
        for(char c: s.toCharArray()){
            // 1. 이전에 들어있는 문자 확인하기
            if(stack.isEmpty()){ // 1-1. 비어있으면, push
                stack.push(c);
            }else{ // 1-2. 비어있지 않으면, peek 
                if(stack.peek() == c){ // 2-1. 현재 문자와 동일하면, 꺼내기
                    stack.pop();
                    continue;
                }else{ // 2-2. 현대 문자와 다르면, 넣기
                    stack.push(c);
                }
            }
        }

        
        return stack.isEmpty() ? 1 : 0;
    }
}